tests/test_collection.py

```diff
@@ -271,6 +271,22 @@ def test_from_invalid_dict_raises_exception(self) -> None:
         with self.assertRaises(pystac.STACTypeError):
             _ = pystac.Collection.from_dict(catalog_dict)
 
+    def test_clone_preserves_assets(self) -> None:
+        path = TestCases.get_path("data-files/collections/with-assets.json")
+        original_collection = Collection.from_file(path)
+        assert len(original_collection.assets) > 0
+
+        cloned_collection = original_collection.clone()
+
+        original_assets_dict = {
+            k: asset.to_dict() for k, asset in original_collection.assets.items()
+        }
+        cloned_assets_dict = {
+            k: asset.to_dict() for k, asset in cloned_collection.assets.items()
+        }
+
+        self.assertDictEqual(original_assets_dict, cloned_assets_dict)
+
```

Comment:
Should probably put a regression test in for `Item` as well?

Reply:
We sort of implicitly test that the assets don't get dropped [here](https://github.com/stac-utils/pystac/blob/a6fffa59f4c6b6433858a8569b35647798593976/tests/test_item.py#L224-L232), but it would probably be good to make it more explicit. I can also update the Collection test to check that the owner is set properly to match the Item test.

Reply:
Updated both tests to make them match in dcd21c0f0c73b4fdf078e2842e9a94cb2cbb6452.